src/main/java/resources/NasaResource.java

```diff
@@ -0,0 +1,55 @@
+package resources;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.QueryParam;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.Response.Status;
+import millom.sandbox.CustomException.InvalidWeatherException;
+import millom.sandbox.service.WeatherService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/nasaapi")
+public class NasaResource {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(NasaResource.class);
+
+  private final WeatherService weatherService;
+
+  public NasaResource(WeatherService weatherService) {
+    this.weatherService = weatherService;
+  }
+
+  public NasaResource() {
```

Comment:
Not needed if `MessaginService` is injected.

Reply:
removed after injecting WeatherService.